src/ThreeEditor/Simulation/Materials/SimulationMaterial.ts

```
@@ -123,7 +128,13 @@ export default class SimulationMaterial extends THREE.MeshPhongMaterial {
 		};
 	}
 
-	static fromJSON(
+	fromSerialized(state: SimulationMaterialJSON) {
+		console.error('SimulationMaterial: deserialization unsupported');
```

why not supported ?

---

there were some cases where fromJSON was not present because it was not used. Instead some classes are rebuild using static version. Here is one of these examples. I wanted it to show an error if someone uses the instance method by mistake, because it would fail silently otherwise.

I'm going to add the same thing to other empty `fromSerialized()`